### PR TITLE
Add streaming support for files endpoint

### PR DESCRIPTION
**Pull request by dileepfrog**

### CODE REVIEW

**Comment by kfang** [comment]
hmm... if you're going to use axios, do we still need nice-request? We should prob standardize on which http request lib the project uses.

**Comment by dileepfrog** [comment]
Axios is clearly superior in every respect, however, I wanted to just enable my feature while introducing an incremental upgrade route rather than pushing the risk onto currently used endpoints all at once. My feelings are we should keep the scope narrow here, see how it works out, then standardize. Thoughts?

File: package.json [hunk 1]
```diff
@@ -1,10 +1,11 @@
 {
   "name": "@headspace/zuora-request-service",
-  "version": "1.1.2-rc0",
+  "version": "1.1.3-rc0",
```
Comment (phubars) [inline package.json]: We should try to avoid incrementing the minor version if we're going to do release candidate builds like `rc0`, `rc1`, etc

**Review by phubars — CHANGES_REQUESTED** [review]
Can we avoid adding another library here if its not being used for anything more specific than making requests? Its seems theres not that much value added.

Also I'm having a hard time understanding the requirement for streaming? Can't we just provide an endpoint for the frontend that uses the mime type `application/pdf`?

**Comment by dileepfrog** [comment]
The current zuora lib is built on nice-request which is built on request-promise. In addition to providing an appropriate mime type the new endpoint must also stream the binary data obtained from the zuora files API. Streaming will not work in request-promise so we cannot use nice-request for this purpose either. We cannot simply provide a link to the files API endpoint as the client does not have zuora API credentials. So I am solving this by using axios which does support streaming and is currently the most widely used NPM for requests (over 43k stars compared to 0 for nice-request).

**Comment by phubars** [comment]
Should be noted that nice-request is simply a wrapper around request promise which is just another wrapper around request https://github.com/request/request, its not a libarary without merit.

Would it be easier to read the bytes from the invoice pdf into memory and then spit it back out to the client the way the v2 api does? https://github.com/HeadspaceMeditation/api/blob/master/src/main/java/com/headspace/api/rest/service/subscription/SubscriptionServiceImpl.java#L2440

**Comment by dileepfrog** [comment]
Ok, will implement with attachment instead of chunked transfer

**Comment by dileepfrog** [comment]
Reopening this after encountering more friction trying to make `nice-request` work here. The latter forces a json response type and supports only a subset of options for `request`. https://github.com/BobbieBarker/nice-request/blob/master/make-request/get-request-options.js#L17
Really not worth the effort to patch up `nice-request` it seems to me as for this special case of binary files `axios` works right out of the box

**Review by kfang — APPROVED** [review]
can we leave a TODO somewhere to remove the use of nice-request somewhere? that way everything just uses axios and we can remove the dependency on nice-request.

**Review by phubars — APPROVED** [review]

**Pull request merged**